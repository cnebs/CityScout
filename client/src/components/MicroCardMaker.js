import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CardActionArea from '@material-ui/core/CardActionArea';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import { Grow } from '@material-ui/core/';
import Typography from '@material-ui/core/Typography';
import ListItem from '@material-ui/core/ListItem';

const useStyles = makeStyles({
  card: {
    maxWidth: '100%',
    minWidth: '100%',
  },
  media: {
    height: '130px',
  },
  content: {
    height: '50px',
  },
  listitem: {
    square: true,
  }
});

export default function MicroCardMaker({ event, handleMicroCardClick , animationTimeout}) {
  const classes = useStyles();
  const eventStart = Date(event.time_start)


  return (
    <ListItem className={classes.listitem} alignItems='flex-start' onClick={() => (handleMicroCardClick(event))}>
      <Grow in={true} timeout={animationTimeout}>
        <Card className={classes.card}>
          <CardActionArea>
            <CardMedia
              className={classes.media}
              image={event.img || 'Event Image'}
              title={event.name || 'Hello, there!'}
            />
            <CardContent className={classes.content}>
              <Typography gutterBottom variant='h5' component='h4'>
                {
                  event.name.length < 45 ?
                    event.name :
                    ((event.name).substring(0, 45) + '...')
                }
              </Typography>
              {/* <Typography variant='body2' color='textSecondary' component='p'>
                Starting at {eventStart.split('GMT')[0].split(' ')[4].split(':').slice(0, 2).join(':') || 'starting time'}
              </Typography> */}
            </CardContent>
          </CardActionArea>
        </Card>
      </Grow>
    </ListItem>
  )
}
