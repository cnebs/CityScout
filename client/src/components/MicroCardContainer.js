import React from 'react';
import MicroCardMaker from './MicroCardMaker.js'
import { makeStyles } from '@material-ui/core/styles';
import { Grid } from '@material-ui/core/';
import Paper from '@material-ui/core/Paper';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import List from '@material-ui/core/List';

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper,
  },
  cardColumn: {
    padding: 10
  }
}));

export default function MicroCardContainer({ eventsToday, eventsTomorrow, eventsTomorrowPlusPlus }) {
  const classes = useStyles();
  const [value, setValue] = React.useState(0);

  function handleChange(event, newValue) {
    setValue(newValue);
  }
  console.log(eventsToday);
  return (

    // Build mini navbar onto container

    <Grid item xs={4}>
      <Paper className={classes.root}>
        <Tabs
          value={value}
          onChange={handleChange}
          indicatorColor='primary'
          textColor='primary'
          centered
        >
          <Tab label='Today' />
          <Tab label='Tomorrow' />
          <Tab label='Overmorrow' />
        </Tabs>
      </Paper>

      <List className={classes.root}>
        {
          eventsToday.map( event => <MicroCardMaker key={event.name} event={event} />)
        }
      </List>
    </Grid>

  );

}
      // {/* <Grid item xs={12} className={classes.cardColumn}>
      // </Grid>
      // <Grid item xs={12} className={classes.cardColumn}>
      //   <MicroCardMaker event={events[0]} />
      // </Grid>
      // <Grid item xs={12} className={classes.cardColumn}>
      //   <MicroCardMaker event={events[0]} />
      // </Grid>
      // <Grid item xs={12} className={classes.cardColumn}>
      //   <MicroCardMaker event={events[0]} />
      // </Grid> */}