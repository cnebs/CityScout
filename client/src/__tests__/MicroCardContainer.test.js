import { shallow } from 'enzyme';
import React from 'react';
import MicroCardContainer from '../components/MicroCardContainer';
import MicroCardMaker from '../components/MicroCardMaker';
import { Grid } from '@material-ui/core/';

describe('MicroCardContainer component', () => {
  let wrapper;
  let event = [
    {
      source_API: 'TicketMaster',
      name: 'Hayes Carll',
      url:
        'http://www.ticketsnow.com/InventoryBrowse/TicketList.aspx?PID=2718472',
      event_id: 'Z7r9jZ1Aejbow',
      time_start: '2019-08-11T02:00:00Z',
      time_end: null,
      category: 'Music',
      image:
        'https://s1.ticketm.net/dam/a/fc1/e7affb5a-4ba1-4e6f-8aad-29c79f4a6fc1_68981_RECOMENDATION_16_9.jpg',
      venue: 'Gruene Hall',
      location: 'New Braunfels',
      price_min: null,
      price_max: null,
      description: null
    }
  ];
  let day = 'testing day';

  beforeAll(() => {
    wrapper = shallow(<MicroCardContainer events={event} />);
  });

  test('it renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('it renders the correct child components', () => {
    expect(wrapper.find(MicroCardMaker).length).toBeGreaterThan(0); // should hold only 1 card
    expect(wrapper.find(Grid).length).toBeGreaterThan(0);
  });
});
