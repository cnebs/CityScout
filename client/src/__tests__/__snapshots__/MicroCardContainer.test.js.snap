// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MicroCardContainer component it renders without crashing 1`] = `undefined`;
